### DISCUSSION TITLE
Installation fails on Docker in Synology NAS

### DISCUSSION THREAD

**Discussion by leonbohmann** [firefly-iii/firefly-iii#4878] [Firefly III] [answered]
I am running Firefly III latest docker version and am trying to set it up on Docker.
After running the image and accessing it via "server.local:7080" I get the following error.

I tried using the public IP of the database, but that didnt work. It seems like the database does not get filled with tables or anything..

**Description**
Could not poll the database: SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo failed: Name or service not known (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) 

**Extra info**
I am running Docker on a Synology NAS. I am starting the instance with:

```
docker run -d -v /volume1/docker/firefly:/var/www/html/storage/upload \
    -p 7080:8080 \
    -e APP_KEY=H0fHYeECYSMO4TRnD8Ukz5L059P2CAh0 \
    -e DB_HOST=localhost \
    -e DB_PORT=3306 \
    -e DB_CONNECTION=mysql \
    -e DB_DATABASE=fireflydb \
    -e DB_USERNAME=fireflyuser \
    -e DB_PASSWORD=fireflypw \
    --name 'fireflyIII' \
        fireflyiii/core:latest
```

See attached the log of the docker container.

**Bonus points**

- I have read the FAQ at https://bit.ly/FF3-FAQ
- I added a screenshot 
- 
![grafik](https://user-images.githubusercontent.com/13386367/123514429-50db0680-d693-11eb-9bf7-1ada6785f525.png)

- I added log files
[fireflylog.txt](https://github.com/firefly-iii/firefly-iii/files/6720157/fireflylog.txt)

**Comment by JC5** [comment]
So you have no database container?

**Reply by leonbohmann** [reply]
Sure I do, it is running on localhost:3306. I created the database called 'fireflydb' with MySQL Workbench!

**Reply by terr-steak** [reply]
Did you grant and flush permissions to the user on the db?

**Reply by leonbohmann** [reply]
Yes I did. What I didnt try is to grant admin privileges, but that shouldnt be the issue here, I suppose?

**Reply by terr-steak** [reply]
I granted `All` to the firefly user on fireflydb.* I am not sure if that is the same as admin (again DB dumb, just trying to help cause I was having similar issues). But it appears to me that the name resolution is not working either so I am also wondering if you should put the ip for the DB server not local host??? Maybe local host is making it think the DB is running in the same container (IP)???

**Reply by JC5** [reply]
It can’t be localhost because that points to firefly iii itself.

**Reply by leonbohmann** [reply]
That might very well be.
How do I use the localhost to the server not the Docker?

**Reply by JC5** [reply]
Not sure. DB_HOST should be the host name of the db container or the internal IP address.

**Reply by terr-steak** [reply]
its either your synology IP or the IP of where youre hosting the DB (if its a docker container then that IP if its a physical server it that machines IP). If you're not sure where to find that IP info let me know and I might be able to help


Another issue I was having (setting it up on unraid) was that when I tried usign DataGrip (sqlWorkbench like tool) it was appending @<hostname> to the user even if I defined it on creation so it was sending <user>@<host I entered><@host dataGrip appended> and I dont think the DB liked that. DB's arent my strong suite. But so I would check the logs and see if maybe the same is happening to you via workbench???

**Comment by terr-steak** [comment, marked as answer]
its either your synology IP or the IP of where youre hosting the DB (if its a docker container then that IP if its a physical server it that machines IP). If you're not sure where to find that IP info let me know and I might be able to help

**Comment by leonbohmann** [comment]
Alright, that did the trick. I was using `server.local` which somehow could not be resolved by Firefly. When using the IP directly via `192.168.0.144` it worked. The db was created correctly and I can use firefly now.

**Reply by hoangduy1505** [reply]
I am a new user of Synology, I run container firefly but it not run
Can you share with me some picture of config varible?

**Reply by Werfjes** [reply]
I am having the same issue. I tried to add the ipnumber of de DB in the Environment variables of the docker. (like: DB_HOST 192.168.88.199) but that didnt work
![FireFly](https://user-images.githubusercontent.com/91469589/134909541-e4e3ec56-19ea-4ea9-9a5a-8253fbcfd8b3.JPG)
